Bottom: cbcc5594a39e9b709b2d5eade559c917213b9aaa
Top:    c39b93b2be7909282dd71feb1f853a09d942acc7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 16:56:13 +0100

draw.c: Change DrawTop() to DrawOuter() taking a side parameter


---

diff --git a/src/draw.c b/src/draw.c
index 8178e81..caf14c7 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -83,7 +83,7 @@ static const BoxType *clip_box = NULL;
  */
 static void Redraw (bool, BoxTypePtr);
 static void DrawEverything (BoxTypePtr);
-static void DrawTop (const BoxType *);
+static void DrawOuterLayer (int side, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void DrawPinOrViaLowLevel (PinTypePtr, bool);
 static void DrawPlainPin (PinTypePtr, bool);
@@ -313,11 +313,11 @@ PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
   int save_swap = SWAP_IDENT;
 
   gui->set_draw_faded (Output.fgGC, 1);
-  SWAP_IDENT = swap_ident;
   DrawLayerGroup (side_group, drawn_area);
-  DrawTop (drawn_area);
+  DrawOuterLayer (side, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
+  SWAP_IDENT = swap_ident;
   /* draw package */
   DrawSilk (swap_ident,
 	    swap_ident ? solder_silk_layer : component_silk_layer,
@@ -410,9 +410,11 @@ DrawEverything (BoxTypePtr drawn_area)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
     return;
 
+  side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
+
   /* Draw pins, pads, vias below silk */
   if (gui->gui)
-    DrawTop (drawn_area);
+    DrawOuterLayer (side, drawn_area);
   else
     {
       HoleCountStruct hcs;
@@ -577,10 +579,8 @@ pad_callback (const BoxType * b, void *cl)
  * draws pins pads and vias
  */
 static void
-DrawTop (const BoxType * screen)
+DrawOuterLayer (side, const BoxType * screen)
 {
-  int side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
-
   if (PCB->PinOn || doing_assy)
     {
       /* draw element pins */
